test/spec/style_layer.js

```diff
@@ -0,0 +1,90 @@
+        server = sinon.fakeServer.create();
+    });
+
+    afterEach(function() {
+        server.restore();
+    });
+
+    describe('Constructor', function() {
+
+        it('sets attribution', function() {
+            var layer = L.mapbox.styleLayer('mapbox://styles/mapbox/streets-v8');
+            layer.on('ready', function(e) {
+                expect(layer.options.attribution).to.equal('<a href="https://www.mapbox.com/about/maps/" target="_blank">&copy; Mapbox</a> <a href="http://www.openstreetmap.org/about/" target="_blank">&copy; OpenStreetMap</a> <a class="mapbox-improve-map" href="https://www.mapbox.com/map-feedback/" target="_blank">Improve this map</a>');
+            });
+        });
+
+        it('sets attribution', function() {
+            var layer = L.mapbox.styleLayer('mapbox://styles/mapbox/hybrid-streets-v8');
+            layer.on('ready', function(e) {
+                expect(layer.options.attribution).to.equal('<a href="https://www.mapbox.com/about/maps/" target="_blank">&copy; Mapbox</a> <a href="http://www.openstreetmap.org/about/" target="_blank">&copy; OpenStreetMap</a> <a class="mapbox-improve-map" href="https://www.mapbox.com/map-feedback/" target="_blank">Improve this map</a> <a href="https://www.digitalglobe.com/" target="_blank">&copy; DigitalGlobe</a>');
+            });
+        });
+
+        it('tms option is always false', function() {
+            var layer = L.mapbox.styleLayer('mapbox://styles/mapbox/streets-v8', {
+                tms: true
+            });
+            layer.on('ready', function(e) {
+                expect(layer.options.tms).to.equal(false);
+            });
+        });
+
+        it('tileSize is always 512', function() {
+            var layer = L.mapbox.styleLayer('mapbox://styles/mapbox/streets-v8', {
+                tileSize: 256
+            });
+            layer.on('ready', function(e) {
+                expect(layer.options.tileSize).to.equal(512);
+            });
+        });
+
+        it('zoomOffset is always -1', function() {
+            var layer = L.mapbox.styleLayer('mapbox://styles/mapbox/streets-v8', {
+                zoomOffset: 0
+            });
+            layer.on('ready', function(e) {
+                expect(layer.options.zoomOffset).to.equal(-1);
+            });
+        });
+
+        it('sanitizes attribution', function() {
+            var layer = L.mapbox.styleLayer('mapbox://styles/mapbox/streets-v8', {attribution: '<script>alert("test")</script>'});
+            layer.on('ready', function(e) {
+                expect(layer.options.attribution).to.equal('');
+            });
+        });
+    });
+
+    describe('#getTileUrl', function() {
+        var retina;
+
+        beforeEach(function() {
+            retina = L.Browser.retina;
+            L.Browser.retina = false;
+        });
+
+        afterEach(function() {
+            L.Browser.retina = retina;
+        });
+
+        it('distributes over the URLs in the tiles property', function() {
```

Comment:
On second thought, we don't want to hard code `a.tiles.mapbox.com` and `b.tiles.mapbox.com`, so we only have `config.HTTPS_URL`. Let's leave the current behavior but change the test description to be more accurate.